### PR TITLE
Extension of referral-table and big-stat for classic use

### PR DESCRIPTION
**Pull request by Locrian24**
### big-stat
- Removes the `programId` requirement for `big-stat` stat queries, and enforces not filtering stats by only Classic.

### referral-table
- Adds "classic" option to `programId` prop to filter on classic only referrals. Also adds the ability to hide column labels and style next/prev buttons
- Add support to show the referrer, the user who referred the current user. For this to work, all table columns must implement the method `renderReferrerCell`. Currently, the columns included with `mint-components` do not implement this functionality.

### Everything else
- The version of `component-boilerplate` use by `mint-components` has been updated to support classic programs. 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: (PUT IT HERE)
 - Process.st launch checklist: https://app.process.st/runs/Hungry-Harvest-Classic-Widget-Update-mNrUitfOWDalDfi1ffxKbg/tasks/uRcCQFmLhyRs07APcFZEEw

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/mint-components/package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
     "workbox-build": "4.3.1"
   },
   "dependencies": {
-    "@saasquatch/component-boilerplate": "^1.2.0",
+    "@saasquatch/component-boilerplate": "^1.2.2-0",
     "@saasquatch/stencil-grapes-plugin": "^1.0.1",
     "@saasquatch/stencil-hooks": "^2.0.1-0",
```
Comment (00salmon) [inline packages/mint-components/package.json:59]:
```suggestion
    "@saasquatch/stencil-hooks": "^2.0.2",
```

File: packages/mint-components/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.1.2] - 2021-08-20
+
+- \<sqm-referral-table> can pull only classic referrals
+- \<sqm-referral-table> supports showing "Referred You" row
+- \<sqm-big-stat> will return appropriate stats or error for "classic" program id
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:14]: Height changes should be noted.

```suggestion
- \<sqm-big-stat> will return appropriate stats or error for "classic" program id. Height is inherited
```

File: packages/mint-components/src/index.html [hunk 3]
```diff
@@ -11,6 +11,11 @@
     <script nomodule src="build/mint-components.js"></script>
   </head>
   <body>
-    <sqm-stencilbook></sqm-stencilbook>
+    <!-- <sqm-stencilbook></sqm-stencilbook> -->
+    <sqm-referral-table>
+      <sqm-referral-table-user-column></sqm-referral-table-user-column>
+      <sqm-referral-table-status-column></sqm-referral-table-status-column>
+      <sqm-referral-table-rewards-column></sqm-referral-table-rewards-column>
+    </sqm-referral-table>
```
Comment (loganvolkers) [inline packages/mint-components/src/index.html:19]: @Locrian24 Never commit this! Add a story with this HTML content instead.

```suggestion
    <sqm-stencilbook></sqm-stencilbook>
```

File: packages/mint-components/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/mint-components",
-  "version": "1.1.1",
+  "version": "1.1.2-12",
```
Comment (loganvolkers) [inline packages/mint-components/package.json:3]: Need to update to `1.1.2`.

```suggestion
  "version": "1.1.2-12",
```

File: packages/mint-components/CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.1.2] - 2021-08-20
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:10]: Need to update to `1.2.0` since this adds some new functionality. (see https://semver.org/)

**Review by loganvolkers — APPROVED** [review]
Looks good to me, but should be published as `1.2.0` since it adds bigger functionality (referred you stuff)

File: packages/mint-components/package.json [hunk 6]
```diff
@@ -54,9 +54,9 @@
     "workbox-build": "4.3.1"
   },
   "dependencies": {
-    "@saasquatch/component-boilerplate": "^1.2.1",
+    "@saasquatch/component-boilerplate": "^1.2.2-0",
```
Comment (loganvolkers) [inline packages/mint-components/package.json:57]: Need to update to `1.2.2`

**Review by loganvolkers — CHANGES_REQUESTED** [review]
Need to wait for this deploy first: https://github.com/saasquatch/program-tools/pull/169

File: packages/mint-components/CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.2.0] - 2021-08-20
```
Comment (00salmon) [inline packages/mint-components/CHANGELOG.md:10]:
```suggestion
## [1.2.0] - 2021-08-31
```

**Review by 00salmon — DISMISSED** [review]
👍

File: packages/mint-components/package.json (hunk 6 above)
Comment (loganvolkers) [reply packages/mint-components/package.json:57]: Still needs to be updated

Comment (00salmon) [inline packages/mint-components/package.json:57]:
```suggestion
    "@saasquatch/component-boilerplate": "^1.2.2",
```

Comment (00salmon) [reply packages/mint-components/package.json]: done!

File: packages/mint-components/package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/mint-components",
-  "version": "1.1.1",
+  "version": "1.1.2-13",
```
Comment (loganvolkers) [inline packages/mint-components/package.json:3]: Should be 1.2.0

**Review by loganvolkers — COMMENTED** [review]
Looks good to me! Needs the NPM and Changelog versions and dates properly set, then can be deployed.

File: packages/mint-components/CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.2.0] - 2021-09-17
+
+- \<sqm-referral-table> can pull only classic referrals
+- \<sqm-referral-table> supports showing "Referred You" row
+- \<sqm-referral-table> supports additional props to override text values
+- \<sqm-big-stat> will return appropriate stats or error for "classic" program id. Height is inherited
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:15]:
```suggestion
- \<sqm-referral-table> can pull only classic referrals
- \<sqm-referral-table> supports showing "Referred You" row
- \<sqm-referral-table> supports additional props to override text values
- \<sqm-referral-table> has default loading and empty slots
- \<sqm-referral-table> pagination button labels are customizable
- \<sqm-share-button> text slot (default slot) is hideable
- \<sqm-big-stat> will return appropriate stats or error for "classic" program id. Height is inherited
```

**Review by loganvolkers — APPROVED** [review]
Looks good to me

**Pull request merged**